app/models/AuthorityResource.java

```diff
@@ -187,6 +187,33 @@ public String gndRelationEdges() {
 		return Json.toJson(result).toString();
 	}
 
+	public Integer getBirthYear() {
+		String[] parts = fieldValues("dateOfBirth-dateOfDeath", json)
+			.collect(Collectors.joining())
+			.split("–");
+		return Integer.parseInt(parts[0]);
+	}
+
+	public Integer getDeathYear() {
+		String[] parts = fieldValues("dateOfBirth-dateOfDeath", json)
+			.collect(Collectors.joining())
+			.split("–");
+		if (parts.length > 1 && parts[1].length() == 4) {
+			try {
+				return Integer.parseInt(parts[1]);
+			} catch (NumberFormatException e) {
+				return null;
+			}
+		}
+		return null;
+	}
+
+	public String getFirstAndLastName() {
+		String[] lastAndFirstName = preferredName.split(", ");
+		String firstAndLastName = lastAndFirstName[1] + " " + lastAndFirstName[0];
```

Comment:
Here, I forgot to handle the case where the preferredName does not contain a comma in the implementation I pointed you to recently. See e.g. https://test.lobid.org/gnd/1208647903. I later fixed it like this:

```suggestion
		String firstAndLastName = lastAndFirstName.length == 2
				? lastAndFirstName[1] + " " + lastAndFirstName[0]
				: preferredName;
```

See https://github.com/hbz/rppd/pull/4/files for the original fix, which contains an additional test that we should add here too. We could also add corresponding tests for the missing dates and possible cases triggering the NumberFormatException (in addition or instead of the unit tests mentioned above).

Reply:
I added a test which covers the name and missing dates, see https://github.com/hbz/lobid-gnd/pull/429/commits/8526351fae7467cab10196f2ac4d6bb688669860